app.js

```diff
@@ -0,0 +1,116 @@
+
+// On load, hide battlefield
+$("#battlefield").hide();
+
+// SPA click handler on 'Do Battle' button
+$("#do-battle").click(function(evt){
+  if ($("#player1").val() === $("#player2").val()) {
+    alert("Please select two different robot models");
+  } else {
+    $("#fighter-select").hide();
+    $("#battlefield").show();
+    setFighters();
+  }
+})
+
+// Loop through list of fighters to find each selection,
+// assign selection and name to fighter1 and fighter2 object
+function setFighters(){
+  for(f in fighters) {
+    if (fighters[f].name == $("#player1").val()) {
+      $.extend(true,fighter1,fighters[f]);
+      fighter1.customName = $("#player1").parent("label").prev().val();
```

Comment:
That's a fragile stack of chaining. Can you find a more direct way to target this?